pkg/models/load_balancer.go

```diff
@@ -11,25 +11,30 @@ type CreateLoadBalancerRequest struct {
 
 type CreateNetworkLoadBalancerRequest struct {
 	ID                  int                       `json:"-" tf:"id,computed"`
-	Name                string                    `json:"name"`
-	Type                string                    `json:"type"`
-	Description         string                    `json:"description"`
-	NetworkServerID     int                       `json:"networkServerId" tf:"network_server_id"`
-	Enabled             bool                      `json:"enabled"`
-	Visibility          string                    `json:"visibility"`
-	Config              CreateConfig              `json:"config"`
-	ResourcePermissions EnableResourcePermissions `json:"resourcePermission" tf:"resource_permission"`
+	Name                string                    `json:"name" tf:"name"`
+	Type                string                    `json:"type" tf:"lb_type,computed"`
+	Description         string                    `json:"description" tf:"description"`
+	NetworkServerID     int                       `json:"networkServerId" tf:"network_server_id,computed"`
+	Enabled             bool                      `json:"enabled" tf:"enabled"`
+	Config              *CreateConfig             `json:"config" tf:"config,sub"`
+	ResourcePermissions EnableResourcePermissions `json:"resourcePermissions" tf:"group_access"`
 }
 
-type CreateConfig struct {
-	AdminState bool   `json:"adminState" tf:"admin_state"`
-	Loglevel   string `json:"loglevel"`
-	Size       string `json:"size"`
-	Tier1      string `json:"tier1"`
+type EnableResourcePermissions struct {
+	All   bool              `json:"all,omitempty" tf:"all"`
```

Comment:
Please validate do we need omitempty here and take it up in the next PR. omitempty functionality should be checked for all the cases.

Reply:
yes its not needed here removed